apps/lib/opt.c

```diff
@@ -1036,14 +1036,14 @@ int opt_check_rest_arg(const char *expected)
             return 1;
         opt_printf_stderr("%s: Missing argument: %s\n", prog, expected);
         return 0;
-    } else if (expected != NULL) {
-        return 1;
     }
+    if (expected != NULL)
```

Comment:
should we be testing this kind of thing?

Reply:
No idea, this was already in the code.  I'd be surprised if something didn't rely on this behaviour.